src/refdb_fs.c

```diff
@@ -63,6 +66,18 @@ typedef struct refdb_fs_backend {
 	uint32_t direach_flags;
 } refdb_fs_backend;
 
+/*
+ * List of symbolic reference names that are always stored in the
+ * gitdir. This is required such that we are able to correctly
+ * resolve these references via the gitdir instead of the
+ * commondir when working trees are used.
+ */
+static const char *symrefs[] = {
```

Comment:
I've removed this array altogether as there is currently only one per-worktree ref that needs to be handled this way, which is `HEAD`. I've also renamed `is_symbolic_ref` to `is_per_worktree_ref`.